### PR TITLE
Implement Radio Component

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/react-science/issues/474

### CODE REVIEW

**Review by stropitek — COMMENTED** [review]
Thank you, the implementation SGTM. I'd like to have a few design improvements and api additions before we merge this.

CleanShot 2023-08-11 at 10 05 53

Given that we are here implementing the "normal" variant of the component, can you make it look like the middle one? Can you also implement the `variant="small"` prop?

CleanShot 2023-08-11 at 10 11 23

Can you make it such that we don't see the grey outline when the radio is selected?


https://github.com/zakodium-oss/react-science/assets/4118690/8d725758-fa74-4e95-a064-b6ca4cda0cc5

Can you change the color of the outline when the mouse hovers the radio so that it matches the color of a selected radio?

CleanShot 2023-08-11 at 10 14 19

Can you avoid the double border on the radio groups.

Could you also show what happens when there is only one option in the select in the stories?

Can you add the `disabled` property to the `RadioGroup` so that it disables the entire component?

**Comment by wadjih-bencheikh18** [comment]
> Can you avoid the double border on the radio groups.

I don't see this in my browser could you recheck again please

**Comment by wadjih-bencheikh18** [comment]
@stropitek  I did all the improvement you asked thank you

File: src/components/forms/RadioGroup/RadioGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { css } from '@emotion/react';
+import * as RadioGroupRadix from '@radix-ui/react-radio-group';
+import { ReactNode } from 'react';
+
+import { InputVariant } from '../styles';
+
+import { RadioOptionButton } from './RadioOptionButton';
+import { RadioOptionClassic } from './RadioOptionClassic';
+
+export interface ValueLabel {
```
Comment (stropitek) [inline src/components/forms/RadioGroup/RadioGroup.tsx:11]: Please rename to `RadioOption`

File: src/components/forms/RadioGroup/RadioGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  label: ReactNode;
+  disabled?: boolean;
+}
+export interface RadioGroupProps {
+  selected?: ValueLabel;
+  type?: 'classic' | 'button';
+  options?: ValueLabel[];
+  onSelect?: (option: ValueLabel) => void;
+  name?: string;
+  disabled?: boolean;
```
Comment (stropitek) [inline src/components/forms/RadioGroup/RadioGroup.tsx:22]: Can you add a story which shows how this prop works?

**Comment by stropitek** [comment]
> I don't see this in my browser could you recheck again please

To see the problem, don't use the zoom of the browser, use the zoom of storybook.

If you look closely, you'll see the issue at 100% zoom as well.

**Comment by wadjih-bencheikh18** [comment]
> If you look closely, you'll see the issue at 100% zoom as well.

I am not seeing the double border. Here's my result:
image

Could you please let me know which web browser you're using?

File: src/components/forms/RadioGroup/RadioGroup.tsx (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/components/forms/RadioGroup/RadioGroup.tsx:11]: Do you want me to rename the file? or the folder?

**Comment by targos** [comment]
I see a double border with Chrome: 
CleanShot 2023-08-14 at 12 30 09

**Comment by wadjih-bencheikh18** [comment]
> I see a double border with Chrome:

I checked safari version and yes it have double border
I fixed it now, Could you please check?

**Comment by wadjih-bencheikh18** [comment]
> Did you also check on the deployed storybook (https://474-create-radio-component.react-science.pages.dev/) or only locally?

I checked the two

**Comment by wadjih-bencheikh18** [comment]
> Safari is also not working for me. Did you also check on the deployed storybook (https://474-create-radio-component.react-science.pages.dev/) or only locally?

I found that navigators does not accept border less then 1px so I refactored the code to avoid it

File: stories/components/radio.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        onSelect={setOption}
+        {...props}
+      />
+    </ExampleGroup>
+  );
+}
+Control.args = {
+  variant: 'default',
+  type: 'classic',
+  disabled: true,
```
Comment (stropitek) [inline stories/components/radio.stories.tsx:76]:
```suggestion
  disabled: false,
```

**Comment by stropitek** [comment]
The borders look good for me now.

> CleanShot 2023-08-11 at 10 05 53
> 
> Given that we are here implementing the "normal" variant of the component, can you make it look like the middle one? Can you also implement the `variant="small"` prop?

Can you make the button version more like the ones highlighted in the screenshot (less padding)

File: src/components/forms/RadioGroup/RadioGroup.tsx (hunk 1 above)
Comment (stropitek) [reply src/components/forms/RadioGroup/RadioGroup.tsx:11]: Juste the interface

**Comment by wadjih-bencheikh18** [comment]
> Can you make the button version more like the ones highlighted in the screenshot (less padding)

Style updated.

**Comment by stropitek** [comment]
I think font size should be 12px both on button and classic versions of the small variant

Currently classic is 12px but button is 14px

**Comment by wadjih-bencheikh18** [comment]
> Currently classic is 12px but button is 14px

It's fixed now

I used 14 in button style in two variant because it's the case in the ant design

**Review by stropitek — APPROVED** [review]

**Pull request merged**